### PR TITLE
How to build a task manager with angular

### LINKED ISSUE #3385: How to Build a Task Manager with Angular
### Proposed title of the article
How to build a task manager with angular

### Introduction paragraph (2-3 paragraphs):
If you are trying to learn a programming language or framework, it is good practice to build simple tasks which will go a long way to help solidify your knowledge in that language.  
In this article, I will work you through building a simple task manager app (popularly known as a To-do app) using angular a JavaScript framework.
A task manager is used to write, organize and rearrange tasks more efficiently. The task manager we are going to build will allow the user to add a task, mark a task as complete and also delete a task.

### Key takeaways: 
1. How to create and structure an angular project.
2. How to create modules and components in angular.
3. Build a simple application.
4. Save tasks to local storage.

### References: 
https://angular.io/docs

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by edidee**
#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

this closes #3385

### CODE REVIEW

**Comment by jumamiller** [comment]
@edidee, kindly link this PR to an issue.

**Comment by edidee** [comment]
Issue number #3385

File: content/articles/how-to-build-a-task-manager-with-angular/index.md [hunk 1]
```diff
@@ -0,0 +1,468 @@
+# How to build a task manager with angular.
+
+## Introduction
+
+If you are trying to learn a programming language or framework, it is good practice to build simple tasks which will go a long way to help solidify your knowledge in that language.  
+
+In this article, I will work you through building a simple task manager app (popularly known as a Todo app) using [angular](https://angular.io/docs) a JavaScript framework.
```
Comment (jumamiller) [inline content/articles/how-to-build-a-task-manager-with-angular/index.md:7]: This sentence misses punctuation on some parts. Make it as simple as possible.

File: content/articles/how-to-build-a-task-manager-with-angular/index.md [hunk 2]
```diff
@@ -0,0 +1,468 @@
...
+
+
+## What is a Task Manager?
+
+A task manager is used to write, organize and rearrange tasks more efficiently. The task manager we are going to build will allow the user to add a task, mark the task as complete and also delete the task.
```
Comment (jumamiller) [inline content/articles/how-to-build-a-task-manager-with-angular/index.md:12]: This sentence is hard to follow.. something like `A task manager writes, organizes, and rearranges tasks more efficiently. The task manager we will build will allow the user to add a task, mark the task as complete, and delete the task.` Try making them as simple as possible!

File: content/articles/how-to-build-a-task-manager-with-angular/index.md [hunk 3]
```diff
@@ -0,0 +1,468 @@
...
+## Key Takeaways
+1. How to create and structure an angular project.
+2. How to create modules and components in angular.
+3. Build a simple application.
+4. Save tasks to local storage
+
+
+
+## Prerequisite
+
```
Comment (jumamiller) [inline content/articles/how-to-build-a-task-manager-with-angular/index.md:24]: Prerequisites not Prerequisite!

File: content/articles/how-to-build-a-task-manager-with-angular/index.md [hunk 4]
```diff
@@ -0,0 +1,468 @@
...
+Before we get started, here are a few things to note:
+
+- You should have [Node.js](https://nodejs.org/en/) installed on your computer.
+- You should be fairly familiar with [Angular](https://angular.io/docs) and JavaScript.
+
```
Comment (jumamiller) [inline content/articles/how-to-build-a-task-manager-with-angular/index.md:29]: Angular is mainly dealing with TypeScript, so you may add it on this list.

File: content/articles/how-to-build-a-task-manager-with-angular/index.md [hunk 5]
```diff
@@ -0,0 +1,468 @@
...
+Knowledge of [TypeScript](https://www.typescriptlang.org/) is useful but not requisite.
```
Comment (jumamiller) [inline content/articles/how-to-build-a-task-manager-with-angular/index.md:30]: Remove this line, and, condense it on the same line with the one I pointed above.

File: content/articles/how-to-build-a-task-manager-with-angular/index.md [hunk 6]
```diff
@@ -0,0 +1,468 @@
...
+3. [Add task](https://www.dropbox.com/scl/fi/kzjnn0b7y1h3aw4p43wkx/How-to-build-a-task-manager-with-angular..paper?dl=0&rlkey=icow7dspxlq9l21h9m8q4m8sy#:uid=945266686904507102840148&h2=Add-Task-to-the-To-do-list)
+4. [Mark task as done](https://www.dropbox.com/scl/fi/kzjnn0b7y1h3aw4p43wkx/How-to-build-a-task-manager-with-angular..paper?dl=0&rlkey=icow7dspxlq9l21h9m8q4m8sy#:uid=829690951889208030685797&h2=Mark-Task-as-Done)
+5. [Delete task](https://www.dropbox.com/scl/fi/kzjnn0b7y1h3aw4p43wkx/How-to-build-a-task-manager-with-angular..paper?dl=0&rlkey=icow7dspxlq9l21h9m8q4m8sy#:uid=673533273713544444667587&h2=Delete-Task)
+6. [Save to local storage](https://www.dropbox.com/scl/fi/kzjnn0b7y1h3aw4p43wkx/How-to-build-a-task-manager-with-angular..paper?dl=0&rlkey=icow7dspxlq9l21h9m8q4m8sy#:uid=156966732915302471724639&h2=Save-task-to-local-storage)
```
Comment (jumamiller) [inline content/articles/how-to-build-a-task-manager-with-angular/index.md:39]: All attachments should reside on this article. Do not take us to another page to view these images. But am also not sure if we really need those images on top of the article?

**Review by jumamiller — CHANGES_REQUESTED** [review]
@edidee, this is a nice article, but I've a couple of  problems regarding your sentences, >90% have one/two issues. Kindly rework on the article before we proceed with the review!

**Comment by edidee** [comment]
@jumamiller I have made changes

**Comment by ahmadmardeni1** [comment]
Please always use the template provided by us.
I edited your initial comment and added the template. Check what matches your article.

**Comment by ahmadmardeni1** [comment]
Plagiarism check done ✅

**Comment by hectorkambow** [comment]
@edidee 

Keeping in mind that if NOT all the checkboxes are met - the PR will be closed 👍

**Comment by jumamiller** [comment]
> @edidee
> 
> Keeping in mind that if NOT all the checkboxes are met - the PR will be closed +1

@edidee treat this as urgent before we move forward with review!

File: content/articles/how-to-build-a-task-manager-with-angular/index.md [hunk 7]
```diff
@@ -0,0 +1,478 @@
...
+
+## Overview
+1. Create a new angular project
+2. Create a component
+3. Add task
+4. Mark task as done
+5. Delete a task
+6. Save to local storage
+
+
```
Comment (jumamiller) [inline content/articles/how-to-build-a-task-manager-with-angular/index.md:36]: This overview should appear immediately after introduction. Use the correct numbering for this part.

**Review by jumamiller — CHANGES_REQUESTED** [review]
Kindly refer to [this](https://github.com/section-engineering-education/engineering-education/pull/2741/files) article; for you got get an overview of an article general structure. You're free to check other resources too.

**Comment by jumamiller** [comment]
Hello @edidee, you didnt restructure the article as had been pointed earlier, are you facing any issues? Did you refer to the link I mentioned above? Kindly read the past articles to get more in-depth overview on how articles are structured.

- Once again , visit [this](https://github.com/section-engineering-education/engineering-education/tree/master/new_contributors) link to get more details.In case you're not able to fix, reach me on slack or any other team member to get help!
>Kindly note failure to follow the set guidelines might lead to article closure!

**Comment by edidee** [comment]
Hi @jumamiller 
I have reached out to you on slack but no response.
I have restructured my article following the guide you gave.

File: content/articles/how-to-build-a-task-manager-with-angular/index.md [hunk 8]
```diff
@@ -0,0 +1,473 @@
+# How to build a task manager with angular.
+
+### Introduction
+As developers, it is common practice to build simple applications. This will help solidify one's knowledge about a particular programming language or framework.
```
Comment (jumamiller) [inline content/articles/how-to-build-a-task-manager-with-angular/index.md:4]: Give a proper intro here....we're specific to Task manager in Angular!
> Use the current version of the article so as not interfere with my review. FYI

File: content/articles/how-to-build-a-task-manager-with-angular/index.md [hunk 9]
```diff
@@ -0,0 +1,473 @@
+# How to build a task manager with angular.
+
+### Introduction
+As developers, it is common practice to build simple applications. This will help solidify one's knowledge about a particular programming language or framework. 
+
+In this article, we will build a simple task manager application (AKA To-do app) using [Angular](https://angular.io/docs). Angular is a JavaScript framework.
```
Comment (jumamiller) [inline content/articles/how-to-build-a-task-manager-with-angular/index.md:6]: Could you rewrite this sentence? I feel like 'Angular is a JavaScript framework.' doesn't fit here after giving reader(s) the link.

File: content/articles/how-to-build-a-task-manager-with-angular/index.md [hunk 10]
```diff
@@ -0,0 +1,473 @@
...
+-  Introduction
+-  Key takeaways
+-  What is a Task Manager
+-  Create a new angular project
+-  Create a component
+-  Add task
+-  Mark task as done
+-  Delete a task
+-  Save to local storage
+-  Summary
```
Comment (jumamiller) [inline content/articles/how-to-build-a-task-manager-with-angular/index.md:19]: There is a way to link headings. For example,

- [Prerequisites](#prerequisites) => [Prerequisites] then concatenate with (#prerequisites).Check markdown docs if you're stuck!

File: content/articles/how-to-build-a-task-manager-with-angular/index.md [hunk 11]
```diff
@@ -0,0 +1,473 @@
...
+
+### What is a Task Manager?
+A task manager writes, organizes, and rearranges tasks more efficiently. The task manager we will build will allow the user to add a task, mark the task as complete, and delete the task.
+
+
+### Create a new angular project
+Before creating a new angular project, one will need to install angular CLI. Run `npm install -g @angular/cli` on the terminal or command prompt to install the CLI.
+
+   ```npm install -g @angular/cli
+   ```
```
Comment (jumamiller) [inline content/articles/how-to-build-a-task-manager-with-angular/index.md:44]: This is a command. You should have something like:
 ```bash 
some text goes here 
```
This can be achieved by adding ```bash then closing it with the same. Check markdown docs.

File: content/articles/how-to-build-a-task-manager-with-angular/index.md [hunk 12]
```diff
@@ -0,0 +1,473 @@
...
+
+After installing the CLI, you can now create a new angular project. Run the CLI command `ng new *project name*` to do this. We will use `todoApp` as the project name for this article.
+
+    ``` ng new todoApp 
+    ```
+ Accept the defaults by pressing the enter key.
+
+The Angular CLI will install the required Angular npm packages and other dependencies for the project.
+
```
Comment (jumamiller) [inline content/articles/how-to-build-a-task-manager-with-angular/index.md:53]: Remember you've instructed reader to install Angular. This next line says "will" while you have already installed??Kindly work on the general flow of the article. This is applied across@

File: content/articles/how-to-build-a-task-manager-with-angular/index.md [hunk 13]
```diff
@@ -0,0 +1,473 @@
...
+The Angular CLI has a server to serve and build applications locally.
```
Comment (jumamiller) [inline content/articles/how-to-build-a-task-manager-with-angular/index.md:54]: This is not an introduction to Angular tutorial, why are we covering all these? Just go straight to the point.

File: content/articles/how-to-build-a-task-manager-with-angular/index.md [hunk 14]
```diff
@@ -0,0 +1,473 @@
...
+To serve your application to the browser, you will first navigate to the workspace folder. Run `cd todoApp` to do so. 
+Next, run the command `ng serve -o` and wait for some minutes; your application will launch in the browser.
+
```
Comment (jumamiller) [inline content/articles/how-to-build-a-task-manager-with-angular/index.md:57]: These 2 lines could be reduced to one: For example:

```To serve your application to the browser, cd into the project directory and run the following commands:```.
Avoid wordy sentences!

File: content/articles/how-to-build-a-task-manager-with-angular/index.md [hunk 15]
```diff
@@ -0,0 +1,473 @@
...
+- An HTML template: states what is rendered on the page.
+- A Typescript class: defines the behavior of the template.
+- A CSS file: it is used to style the template.
+
+Components can be created manually or by using the Angular CLI. For this article, it will be created using the latter. You can start by running the command below on your terminal.
+
+    ```
+    ng generate component todo
+    
+    // OR
```
Comment (jumamiller) [inline content/articles/how-to-build-a-task-manager-with-angular/index.md:79]: Comments on the shell uses # and not //. Kindly read markdown docs!

File: content/articles/how-to-build-a-task-manager-with-angular/index.md [hunk 16]
```diff
@@ -0,0 +1,473 @@
...
+    ```
+
+The CLI command above tells the Angular CLI to generate a `todo` component.
+Now in your code, check the `app` folder, which is in the `src` folder, and you will notice a `todo` folder has been created. This `todo` folder contains the HTML, CSS, and TypeScript files, as shown below.
+
+![Angular Setup](/how-to-build-a-task-manager-with-angular/angular_setup.png)
+
+
+After generating the component, it will be automatically imported into the app module. The component is also added to the `declarations` array. You should check the `app.module.ts` file to see this.
+
```
Comment (jumamiller) [inline content/articles/how-to-build-a-task-manager-with-angular/index.md:91]: This is not an intro to Angular tutorial, and we assume the reader already have basics in Angular!

**Review by jumamiller — CHANGES_REQUESTED** [review]
Hello @edidee, is this an introduction to Angular article, it teaches 'very basic' features in Angular which I find not helpful based on the topic. Could you rewrite this article?

**Comment by jumamiller** [comment]
Will reopen as soon as you're done with changes.

**Comment by edidee** [comment]
Hi @jumamiller 
I have made adjustments to the article.

**Comment by jumamiller** [comment]
reopening to check new commits

**Comment by jumamiller** [comment]
> reopening to check new commits

A better version! I'll proceed with the review!

File: content/articles/how-to-build-a-task-manager-with-angular/index.md [hunk 17]
```diff
@@ -0,0 +1,452 @@
...
+![Todo List](/engineering-education/how-to-build-a-task-manager-with-angular/todo-list.png)
+
+> Note: The buttons are static, so we will make them work next.
+
+###  Add Tasks to the To-do list
+Tasks will be added to the to-do list by clicking the `Add` button or by pressing the `Enter` key.
+
+Let's see what happens under the hood;
+Since the `Add` button and the input box are in a form element, we handle them using the Angular [FormGroup](https://angular.io/api/forms/FormGroup#description) property.
+
```
Comment (jumamiller) [inline content/articles/how-to-build-a-task-manager-with-angular/index.md:113]: Could explain why we're using the FormGroup and not Formbuilder?  Give us more details about this.

File: content/articles/how-to-build-a-task-manager-with-angular/index.md [hunk 18]
```diff
@@ -0,0 +1,452 @@
...
+To use the FormGroup property, we first import it in the `app.module.ts` file as shown below:
+```typescript
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
+```
+The property will be automatically included in the `imports` array, as shown in the image below.
```
Comment (jumamiller) [inline content/articles/how-to-build-a-task-manager-with-angular/index.md:118]: Is this right? I have never seen this added automatically in the array as you describe it here...

File: content/articles/how-to-build-a-task-manager-with-angular/index.md [hunk 19]
```diff
@@ -0,0 +1,452 @@
...
+
+
+![Form group](/how-to-build-a-task-manager-with-angular/formgroup1.png)
+
+
+In the `todo.component.html`, add the `[FormGroup]` property in the `<form>` element and set it to `newTodoForm`. This will help collect and collate the value of each element into an object. 
+We will also add a `formControlName` and set it to `todoItem`.
+
```
Comment (jumamiller) [inline content/articles/how-to-build-a-task-manager-with-angular/index.md:126]: Rewrite this paragraph, it doesnt have any flow!

File: content/articles/how-to-build-a-task-manager-with-angular/index.md [hunk 20]
```diff
@@ -0,0 +1,452 @@
...
+                <h1 class="text-grey-darkest">Todo List</h1>
+                <form [formGroup]="newTodoForm" class="flex mt-4" (ngSubmit)="addTask()">
+                    <input class="shadow appearance-none border rounded w-full py-2 px-3 mr-4 text-grey-darker" placeholder="Add Todo" formControlName="todoItem">
+                    <button type='submit' class="flex-no-shrink p-2 border-2 rounded text-teal border-teal hover:text-white hover:bg-teal">Add</button>
+                </form> 
+            </div>
+  ```          
+Finally, to make our form functional, we will write some JavaScript code in our `todo.component.ts` file. Follow these procedures; 
+
+
```
Comment (jumamiller) [inline content/articles/how-to-build-a-task-manager-with-angular/index.md:144]: Angular uses TypeScript and NOT JS

File: content/articles/how-to-build-a-task-manager-with-angular/index.md [hunk 21]
```diff
@@ -0,0 +1,452 @@
...
+1. Import the `[FormBuilder](https://angular.io/api/forms/FormBuilder)` service from the `@angular/forms` package. This service gives suitable methods for creating controls.
+
+
+```javascript
```
Comment (jumamiller) [inline content/articles/how-to-build-a-task-manager-with-angular/index.md:148]: Typescript in Angular!!

File: content/articles/how-to-build-a-task-manager-with-angular/index.md [hunk 22]
```diff
@@ -0,0 +1,452 @@
...
+    //todo.component.ts
+    
+    export class TodoComponent implements OnInit {
+    constructor(
+    private formBuilder: FormBuilder
+    ) {}
+    } 
+```
+
+3. We will use the `FormBuilder` `group()` method to collect the inputted task. These methods set the `newTodoForm` property to a form model containing the `todoItem` field.
```
Comment (jumamiller) [inline content/articles/how-to-build-a-task-manager-with-angular/index.md:167]: Minimize 'we will' phrase, you have used it a lot..make your article flow!

**Review by jumamiller — CHANGES_REQUESTED** [review]
I've noticed a couple of flow issues and I have pointed a few. Kindly have a look at them so that I may proceed with tech review..

**Review by jumamiller — APPROVED** [review]
@edidee, ensure you go over to the edits to improve your future submissions!

**Comment by hectorkambow** [comment]
Jumping on this one
